packages/date-picker/src/vaadin-date-picker-mixin.js

```diff
@@ -585,7 +588,7 @@ export const DatePickerMixin = (subclass) =>
     /** @private */
     _parseDate(str) {
       // Parsing with RegExp to ensure correct format
-      const parts = /^([-+]\d{1}|\d{2,4}|[-+]\d{6})-(\d{1,2})-(\d{1,2})$/.exec(str);
+      const parts = /^([-+]\d{1}|\d{2,4}|[-+]\d{6})-(\d{1,2})-(\d{1,2})(T[\d:.]+Z)?$/.exec(str);
```

Comment:
Allowing `Z` will add a notion that the component is timezone aware - which it is not. If you specify a `Z` in an ISO 8601 string, that means that this date string is explicitly in `UTC`. However for most use-cases in a UI, you want to display dates/times in the user's local timezone. Currently we don't deal with this issue at all, because you are not allowed to specify time zones in the date string, and developers have to make sure they convert their date values into a proper local representation, and vice versa. However if we allow pass an ISO 8601 string in UTC to the component, I would expect the date picker to display a local date.

Example:
- I'm in GMT+02
- my backend returns an ISO date string of `2022-05-31T23:00:00.000Z`
- I set that date string as value on the date picker
- the date picker shows a date of `2022-05-31`, even though in my timezone the date is `2022-06-01`

So looking at typical use-cases how dates are handled between frontend and backend applications, this change doesn't really solve the problem that developers will have to wrangle dates manually somehow.

I'm wondering why `_parseDate` does not use the `new Date(dateString)` contructor, which seems to handle this topic reasonable well. Unless you specify a `Z`, the date string is assumed to be in the user's local time zone - which is the same as the current behavior. When you add a `Z`, the date is parsed as a UTC date string, and the resulting date is converted into the user's local time zone.

Reply:
> I'm wondering why _parseDate does not use the new Date(dateString) contructor

Using a string with the `Date` constructor (and `Date.parse`) is [discouraged](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date#several_ways_to_create_a_date_object)

Reply:
I don't think we are giving any false expectations, we want to support the `toISOString` which already says that the only suffix is Z https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString.

But anyways, we are not saying that we support timezones, we are considering always GMT, so if Z is provided is obviously still valid, we fail with any other suffix is there.

Reply:
> we are considering always GMT

Date-picker doesn't know anything about timezones nor GMT.
If we just strip out the Z suffix, the resulting date would be wrong as pointed out in the example case above.

Besides, `toISOString()` includes hours, minutes, seconds and milliseconds and date-picker doesn't operate on that precision so passing a value like that wouldn't make sense. This is also the case with native `<input type="date">`:

![Screenshot 2022-06-21 at 12 40 43](https://user-images.githubusercontent.com/1222264/174769562-2050936e-d39a-4960-a496-72df9f3e789d.png)